### PR TITLE
Kycpubkeylookup

### PR DESCRIPTION
**Pull request by lawlawlaw**
Added some RPCs to facilitate whitelisting/blacklisting

1. getkycpubkey - returns kyc public key given a wallet address.
2. blacklistkycpubkey
3. whitelistkycpubkeys - either declare a unassigned KYC pub key, or rewhitelist a blacklisted one
4. topupkycpubkeys - top up list of unregisterd kycpubkeys

### CODE REVIEW

File: qa/rpc-tests/onboard.py [hunk 1]
```diff
@@ -189,7 +168,7 @@ def run_test (self):
  
         #Node 1 registers additional addresses to whitelist
         nadd=100
-        self.nodes[1].sendaddtowhitelisttx(nadd,"CBT")
+        self.nodes[1].sendaddtowhitelisttx(nadd,"ISSUANCE")
```
Comment (nkostoulas) [inline qa/rpc-tests/onboard.py]: Shouldn't this use the whitelist asset?

File: src/policy/whitelist.cpp [hunk 2]
```diff
@@ -439,14 +481,34 @@ bool CWhiteList::Update(const CTransaction& tx, const CCoinsViewCache& mapInputs
     return false;
 }
 
+//hashTx = tx->GetHash();){
```
Comment (nkostoulas) [inline src/policy/whitelist.cpp]: remove

File: qa/rpc-tests/onboard.py (hunk 1 above)
Comment (lawlawlaw) [reply qa/rpc-tests/onboard.py]: No - this tx is sent from the client node (self registering additional addresses) so they pay tx fee with the issued token.

File: src/policy/whitelist.cpp (hunk 2 above)
Comment (lawlawlaw) [reply src/policy/whitelist.cpp]: Done

File: qa/rpc-tests/onboard.py (hunk 1 above)
Comment (nkostoulas) [reply qa/rpc-tests/onboard.py]: Ok you can still use CBT or any issued token. ISSUANCE is for createrawissuance not sure why it was added in this test.

**Comment by lawlawlaw** [comment]
I have merged the changes done by @AstryX .

**Pull request merged**